expyfun/analyze/_analyze.py

```diff
@@ -42,39 +54,67 @@ def press_times_to_hmfc(presses, targets, foils, tmin, tmax,
     press by this function. However, there is no such de-bouncing of responses
     to "other" times.
     """
+    known_types = ['counts', 'rts']
+    if isinstance(return_type, string_types):
+        singleton = True
+        return_type = [return_type]
+    else:
+        singleton = False
+    for r in return_type:
+        if not isinstance(r, string_types) or r not in known_types:
+            raise ValueError('r must be one of %s, got %s' % (known_types, r))
     # Sanity check that targets and foils don't overlap (due to tmin/tmax)
     targets = np.atleast_1d(targets) + tmin
     foils = np.atleast_1d(foils) + tmin
     dur = float(tmax - tmin)
     assert dur > 0
     presses = np.sort(np.atleast_1d(presses))
     assert targets.ndim == foils.ndim == presses.ndim == 1
-    all_times = np.concatenate(([-np.inf], targets, foils, [np.inf]))
-    order = np.argsort(all_times)
-    inv_order = np.argsort(order)
-    all_times = all_times[order]
-    if not np.all(all_times[:-1] + dur <= all_times[1:]):
+    # Stack as targets, then foils
+    stim_times = np.concatenate(([-np.inf], targets, foils, [np.inf]))
+    order = np.argsort(stim_times)
+    stim_times = stim_times[order]
+    if not np.all(stim_times[:-1] + dur <= stim_times[1:]):
         raise ValueError('Analysis windows for targets and foils overlap')
-    # Let's just loop (could probably be done with vector math, but it's
-    # too hard and unlikely to be correct)
-    locs = np.searchsorted(all_times, presses, 'right')
-    if len(locs) > 0:
-        assert locs.max() < len(all_times)  # should be True b/c of np.inf
-        assert locs.min() >= 1
-
-    # figure out which presses were to target or masker (valid_idx)
-    in_window = (presses <= all_times[locs - 1] + dur)
-    valid_idx = np.where(in_window)[0]
-    n_other = np.sum(~in_window)
-
+    # figure out what targ/mask times our presses correspond to
+    press_to_stim = np.searchsorted(stim_times, presses, 'right') - 1
```

Comment:
is `right` vs. `left` going to make a difference here?  Should only matter if there are multiple elements of `stim_times` that are the same as each other and also happen to exactly match an element of `presses`.